MapboxCoreNavigation/OfflineDirections.swift

```diff
@@ -0,0 +1,116 @@
+
+public enum OfflineRoutingError: Error {
+    case unexpectedRouteResult
+    case corruptRouteData
+    case responseError(String)
+}
+
+struct OfflineDirectionsConstants {
+    static let offlineSerialQueueLabel = Bundle.mapboxCoreNavigation.bundleIdentifier!.appending(".offline")
+    static let serialQueue = DispatchQueue(label: OfflineDirectionsConstants.offlineSerialQueueLabel)
+}
+
+@objc(MBOfflineDirectionsProtocol)
+public protocol OfflineDirectionsProtocol {
+    
+    /**
+     Initializes a newly created directions object with an optional access token and host.
+     
+     - parameter accessToken: A Mapbox [access token](https://www.mapbox.com/help/define-access-token/). If an access token is not specified when initializing the directions object, it should be specified in the `MGLMapboxAccessToken` key in the main application bundle’s Info.plist.
+     - parameter host: An optional hostname to the server API. The [Mapbox Directions API](https://www.mapbox.com/api-documentation/?language=Swift#directions) endpoint is used by default.
+     - parameter tilesPath: The location where the tiles has been sideloaded to.
+     - parameter translationsPath: The location where the translations has been sideloaded to.
+     */
+    init(accessToken: String?, host: String?, tilesPath: String, translationsPath: String, completionHandler: @escaping OfflineDirectionsCompletionHandler)
+    
+    /**
+     Begins asynchronously calculating the route or routes using the given options and delivers the results to a closure.
+     
+     This method retrieves the routes asynchronously via MapboxNavigationNative.
+     
+     Routes may be displayed atop a [Mapbox map](https://www.mapbox.com/maps/). They may be cached but may not be stored permanently. To use the results in other contexts or store them permanently, [upgrade to a Mapbox enterprise plan](https://www.mapbox.com/directions/#pricing).
+     
+     - parameter options: A `RouteOptions` object specifying the requirements for the resulting routes.
+     - parameter completionHandler: The closure (block) to call with the resulting routes. This closure is executed on the application’s main thread.
+     */
+    func calculateOffline(_ options: RouteOptions, completionHandler: @escaping Directions.RouteCompletionHandler)
+}
+
+@objc(MBMapboxOfflineDirections)
+public class MapboxOfflineDirections: Directions, OfflineDirectionsProtocol {
+    
+    required public init(accessToken: String?, host: String?, tilesPath: String, translationsPath: String, completionHandler: @escaping OfflineDirectionsCompletionHandler) {
+        
+        super.init(accessToken: accessToken, host: host)
+        
+        OfflineDirectionsConstants.serialQueue.sync {
+            let tilesPath = tilesPath.replacingOccurrences(of: "file://", with: "")
+            let translationsPath = translationsPath.replacingOccurrences(of: "file://", with: "")
```

Comment:
This makes it look like `tilesPath` and `translationsPath` aren’t actually paths but rather file URL strings. If so, we should stick to the URL type rather than String and call them URLs.